File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lookup = {}
        for idx,i in enumerate(nums):
            print(f"i:{i},idx:{idx}")
            curr = target-i
            if curr in lookup:
                if idx != lookup[curr]:
                    return [lookup[curr],idx]
            else:
                lookup[i]=idx

File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  let lookup = new Map() 
  for (let i = 0; i<nums.length;i++){
     if(lookup.has(target-nums[i])){
        if ( lookup.get(target-nums[i]) !== i){
            return [lookup.get(target-nums[i]),i]
        }
     }
     else{
        lookup.set(nums[i],i)
     }
  }
};